### PR TITLE
[dotnet] Add reflection-based JSON serialization fallback to `Command`s

### LINKED ISSUE #14731: [🐛 Bug]: [dotnet] AOT changes have broken anonymous types
### What happened?

Before the source-generated JSON serializer was introduced, anonymous types would serialize correctly. Now, an exception is thrown.

It's worth noting that there is nothing the source-generator can do to support anonymous types: https://github.com/dotnet/runtime/issues/78164

There needs to be a fallback option.

Potentially found in https://github.com/aquality-automation/aquality-selenium-dotnet/issues/265

### How can we reproduce the issue?

```shell
var options = new ChromeOptions();
options.AddAdditionalOption("selenoid:options", new { enableVNC = true, enableVideo = false });

var driver = new ChromeDriver(options);

// If you go to `Command.s_jsonSerializerOptions` and comment out `TypeInfoResolver = CommandJsonSerializerContext.Default`, the above code works.
```

### Relevant log output

```shell
System.NotSupportedException : JsonTypeInfo metadata for type '<>f__AnonymousType0`2[System.Boolean,System.Boolean]' was not provided by TypeInfoResolver of type 'OpenQA.Selenium.CommandJsonSerializerContext'. If using source generation, ensure that all root types passed to the serializer have been annotated with 'JsonSerializableAttribute', along with any types that might be serialized polymorphically. The unsupported member type is located on type 'System.Object'. Path: $.
  ----> System.NotSupportedException : JsonTypeInfo metadata for type '<>f__AnonymousType0`2[System.Boolean,System.Boolean]' was not provided by TypeInfoResolver of type 'OpenQA.Selenium.CommandJsonSerializerContext'. If using source generation, ensure that all root types passed to the serializer have been annotated with 'JsonSerializableAttribute', along with any types that might be serialized polymorphically.
```

```shell
ThrowHelper.ThrowNotSupportedException(WriteStack& state, NotSupportedException ex)
JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
JsonTypeInfo`1.Serialize(Utf8JsonWriter writer, T& rootValue, Object rootValueBoxed)
JsonSerializer.WriteString[TValue](TValue& value, JsonTypeInfo`1 jsonTypeInfo)
JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
Command.get_ParametersAsJsonString() line 105
HttpRequestInfo.ctor(Uri serverUri, Command commandToExecute, HttpCommandInfo commandInfo) line 372
HttpCommandExecutor.ExecuteAsync(Command commandToExecute) line 182
DriverServiceCommandExecutor.ExecuteAsync(Command commandToExecute) line 125
WebDriver.ExecuteAsync(String driverCommandToExecute, Dictionary`2 parameters) line 606
```
```


### Operating System

Windows 11

### Selenium version

`trunk`

### What are the browser(s) and version(s) where you see this issue?

N/A

### What are the browser driver(s) and version(s) where you see this issue?

N/A

### Are you using Selenium Grid?

No

### PR DESCRIPTION
**Pull request by RenderMichael**
### **User description**
Fixes #14731

**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Adds a reflection-based JSON serialization fallback, preventing unchecked exceptions from breaking scenarios.

### Motivation and Context

Exceptions are being thrown for unknown types, including anonymous types.
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
Bug fix, Tests

___

### **Description**
- Introduced a reflection-based JSON serialization fallback in the `Command` class to handle unknown types, including anonymous types, preventing unchecked exceptions.
- Enhanced JSON serialization by combining `CommandJsonSerializerContext.Default` with `DefaultJsonTypeInfoResolver`.
- Added unit tests to verify the serialization of anonymous types within the `Command` class, ensuring robustness and correctness.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Command.cs</strong><dd><code>Add reflection-based JSON serialization fallback in Command class</code></dd></summary>
<hr>

dotnet/src/webdriver/Command.cs

<li>Added <code>JsonTypeInfoResolver.Combine</code> to JSON serializer options.<br> <li> Included <code>DefaultJsonTypeInfoResolver</code> for enhanced serialization.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14732/files#diff-713ff04e841a84c186ba6acf6d405ad2b3eed93d2e850a3634cf54fb7edb3ae2">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CommandTests.cs</strong><dd><code>Add tests for JSON serialization of Command class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

dotnet/test/common/CommandTests.cs

<li>Added new test class <code>CommandTests</code>.<br> <li> Implemented test for serializing anonymous types in commands.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14732/files#diff-1c58b4921076c552b8954d836fda5e231edf0264e3682b855b37cc3a912fdfdd">+22/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by nvborisenko** [comment]
This is very good! Nice catch!

**Comment by RenderMichael** [comment]
In the short-term, this should stop all the JSON failures people may be having.

In the long-term, this defeats some the purpose of source-generation. It's still faster, better, future-proof, etc. but it makes the AOT support "best effort". Instead, we may need separate overloads for the methods that take `Dictionary<string, object>` as parameters. We can expose an overload that takes a `JsonNode` parameter for those that want to *guarantee* AOT support.

We can also add an extra parameter of `JsonTypeInfo<Dictionary<string, object>>` for users to provide their own type info, for the same goal.

Either way, the `Dictionary<string, object>` parameters can never truly be AOT-safe

**Comment by nvborisenko** [comment]
Short-term: this PR is very good for regular users (99%), for AOT let's review each particular case (if it happens).
Long-term: "classic" commands/responses might be forwarded to BiDi implementation (strongly-typed).

**Comment by RenderMichael** [comment]
AOT is music to my ears :) One day this project can have the `<IsAotCompatible>true</IsAotCompatible>` property and will work fully.

I made another small PR as part of incremental progress https://github.com/SeleniumHQ/selenium/pull/14733

**Review by nvborisenko — APPROVED** [review]
Very good drug from headache!

File: dotnet/test/common/CommandTests.cs
Comment (nvborisenko) [inline dotnet/test/common/CommandTests.cs:1]: CI is failing, probably selenium project requires license header for all files. Can you please add it?

Comment (RenderMichael) [reply dotnet/test/common/CommandTests.cs:1]: Oops, I didn't think of this for the test suite. added.

**Pull request merged**